### ISSUE TITLE
Docker开发环境启动失败

### ISSUE THREAD

**Issue by icyleaf** [tryzealot/zealot#1198]
### Discussed in https://github.com/tryzealot/zealot/discussions/1184

<div type='discussions-op-text'>

<sup>Originally posted by **mlch911** June 27, 2023</sup>
按照[官方教程](https://zealot.ews.im/zh-Hans/docs/contributing-guide/local-development/devcontainer)配置的环境，执行`bin/dev`直接报错

<details><summary>错误日志</summary>
<p>

```shell
vscode ➜ /workspace (develop) $ bin/dev
08:15:54 web.1  | started with pid 993
08:15:54 job.1  | started with pid 994
08:15:54 js.1   | started with pid 995
08:15:54 css.1  | started with pid 996
08:15:54 js.1   | yarn run v1.22.19
08:15:54 css.1  | yarn run v1.22.19
08:15:54 css.1  | $ sass app/assets/stylesheets/application.sass.scss:app/assets/builds/application.css --no-source-map --load-path=node_modules --watch
08:15:54 js.1   | $ esbuild app/javascript/*.* --bundle --sourcemap --outdir=app/assets/builds --public-path=assets --watch
08:15:54 js.1   | /bin/sh: 1: esbuild: not found
08:15:54 css.1  | /bin/sh: 1: sass: not found
08:15:54 css.1  | error Command failed with exit code 127.
08:15:54 css.1  | info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
08:15:54 js.1   | error Command failed with exit code 127.
08:15:54 js.1   | info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
08:15:55 js.1   | exited with code 127
08:15:55 system | sending SIGTERM to all processes
08:15:55 css.1  | exited with code 127
08:15:55 web.1  | [WARNING] Could not load command "rails/commands/server/server_command". Error: SIGTERM.
08:15:55 web.1  | /home/vscode/.asdf/installs/ruby/3.2.0/lib/ruby/gems/3.2.0/gems/bootsnap-1.16.0/lib/bootsnap/compile_cache/iseq.rb:66:in `fetch'
08:15:55 web.1  | /home/vscode/.asdf/installs/ruby/3.2.0/lib/ruby/gems/3.2.0/gems/bootsnap-1.16.0/lib/bootsnap/compile_cache/iseq.rb:85:in `load_iseq'
08:15:55 web.1  | /home/vscode/.asdf/installs/ruby/3.2.0/lib/ruby/gems/3.2.0/gems/tzinfo-2.0.6/lib/tzinfo.rb:42:in `require_relative'
08:15:55 web.1  | /home/vscode/.asdf/installs/ruby/3.2.0/lib/ruby/gems/3.2.0/gems/tzinfo-2.0.6/lib/tzinfo.rb:42:in `<main>'
08:15:55 web.1  | /home/vscode/.asdf/installs/ruby/3.2.0/lib/ruby/gems/3.2.0/gems/bootsnap-1.16.0/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:32:in `require'
08:15:55 web.1  | /home/vscode/.asdf/installs/ruby/3.2.0/lib/ruby/gems/3.2.0/gems/bootsnap-1.16.0/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:32:in `require'
08:15:55 web.1  | /home/vscode/.asdf/installs/ruby/3.2.0/lib/ruby/gems/3.2.0/gems/activesupport-7.0.4.3/lib/active_support/values/time_zone.rb:3:in `<main>'
08:15:55 web.1  | /home/vscode/.asdf/installs/ruby/3.2.0/lib/ruby/gems/3.2.0/gems/bootsnap-1.16.0/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:32:in `require'
08:15:55 web.1  | /home/vscode/.asdf/installs/ruby/3.2.0/lib/ruby/gems/3.2.0/gems/bootsnap-1.16.0/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:32:in `require'
08:15:55 web.1  | /home/vscode/.asdf/installs/ruby/3.2.0/lib/ruby/gems/3.2.0/gems/activesupport-7.0.4.3/lib/active_support/core_ext/time/conversions.rb:5:in `<main>'
08:15:55 web.1  | /home/vscode/.asdf/installs/ruby/3.2.0/lib/ruby/gems/3.2.0/gems/bootsnap-1.16.0/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:32:in `require'
08:15:55 web.1  | /home/vscode/.asdf/installs/ruby/3.2.0/lib/ruby/gems/3.2.0/gems/bootsnap-1.16.0/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:32:in `require'
08:15:55 web.1  | /home/vscode/.asdf/installs/ruby/3.2.0/lib/ruby/gems/3.2.0/gems/activesupport-7.0.4.3/lib/active_support/core_ext/object/json.rb:14:in `<main>'
08:15:55 web.1  | /home/vscode/.asdf/installs/ruby/3.2.0/lib/ruby/gems/3.2.0/gems/bootsnap-1.16.0/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:32:in `require'
08:15:55 web.1  | /home/vscode/.asdf/installs/ruby/3.2.0/lib/ruby/gems/3.2.0/gems/bootsnap-1.16.0/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:32:in `require'
08:15:55 web.1  | /home/vscode/.asdf/installs/ruby/3.2.0/lib/ruby/gems/3.2.0/gems/activesupport-7.0.4.3/lib/active_support/core_ext/object.rb:13:in `<main>'
08:15:55 web.1  | /home/vscode/.asdf/installs/ruby/3.2.0/lib/ruby/gems/3.2.0/gems/bootsnap-1.16.0/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:32:in `require'
08:15:55 web.1  | /home/vscode/.asdf/installs/ruby/3.2.0/lib/ruby/gems/3.2.0/gems/bootsnap-1.16.0/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:32:in `require'
08:15:55 web.1  | /home/vscode/.asdf/installs/ruby/3.2.0/lib/ruby/gems/3.2.0/gems/railties-7.0.4.3/lib/rails/configuration.rb:4:in `<main>'
08:15:55 web.1  | /home/vscode/.asdf/installs/ruby/3.2.0/lib/ruby/gems/3.2.0/gems/bootsnap-1.16.0/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:32:in `require'
08:15:55 web.1  | /home/vscode/.asdf/installs/ruby/3.2.0/lib/ruby/gems/3.2.0/gems/bootsnap-1.16.0/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:32:in `require'
08:15:55 web.1  | /home/vscode/.asdf/installs/ruby/3.2.0/lib/ruby/gems/3.2.0/gems/railties-7.0.4.3/lib/rails/railtie/configuration.rb:3:in `<main>'
08:15:55 web.1  | /home/vscode/.asdf/installs/ruby/3.2.0/lib/ruby/gems/3.2.0/gems/bootsnap-1.16.0/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:32:in `require'
08:15:55 web.1  | /home/vscode/.asdf/installs/ruby/3.2.0/lib/ruby/gems/3.2.0/gems/bootsnap-1.16.0/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:32:in `require'
08:15:55 web.1  | /home/vscode/.asdf/installs/ruby/3.2.0/lib/ruby/gems/3.2.0/gems/railties-7.0.4.3/lib/rails/railtie.rb:264:in `config'
08:15:55 web.1  | /home/vscode/.asdf/installs/ruby/3.2.0/lib/ruby/gems/3.2.0/gems/railties-7.0.4.3/lib/rails/railtie.rb:146:in `config'
08:15:55 web.1  | /home/vscode/.asdf/installs/ruby/3.2.0/lib/ruby/gems/3.2.0/gems/activesupport-7.0.4.3/lib/active_support/i18n_railtie.rb:10:in `<class:Railtie>'
08:15:55 web.1  | /home/vscode/.asdf/installs/ruby/3.2.0/lib/ruby/gems/3.2.0/gems/activesupport-7.0.4.3/lib/active_support/i18n_railtie.rb:9:in `<module:I18n>'
08:15:55 web.1  | /home/vscode/.asdf/installs/ruby/3.2.0/lib/ruby/gems/3.2.0/gems/activesupport-7.0.4.3/lib/active_support/i18n_railtie.rb:8:in `<main>'
08:15:55 web.1  | /home/vscode/.asdf/installs/ruby/3.2.0/lib/ruby/gems/3.2.0/gems/bootsnap-1.16.0/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:32:in `require'
08:15:55 web.1  | /home/vscode/.asdf/installs/ruby/3.2.0/lib/ruby/gems/3.2.0/gems/bootsnap-1.16.0/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:32:in `require'
08:15:55 web.1  | /home/vscode/.asdf/installs/ruby/3.2.0/lib/ruby/gems/3.2.0/gems/activesupport-7.0.4.3/lib/active_support/railtie.rb:4:in `<main>'
08:15:55 web.1  | /home/vscode/.asdf/installs/ruby/3.2.0/lib/ruby/gems/3.2.0/gems/bootsnap-1.16.0/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:32:in `require'
08:15:55 web.1  | /home/vscode/.asdf/installs/ruby/3.2.0/lib/ruby/gems/3.2.0/gems/bootsnap-1.16.0/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:32:in `require'
08:15:55 web.1  | /home/vscode/.asdf/installs/ruby/3.2.0/lib/ruby/gems/3.2.0/gems/railties-7.0.4.3/lib/rails.rb:16:in `<main>'
08:15:55 web.1  | /home/vscode/.asdf/installs/ruby/3.2.0/lib/ruby/gems/3.2.0/gems/bootsnap-1.16.0/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:32:in `require'
08:15:55 web.1  | /home/vscode/.asdf/installs/ruby/3.2.0/lib/ruby/gems/3.2.0/gems/bootsnap-1.16.0/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:32:in `require'
08:15:55 web.1  | /home/vscode/.asdf/installs/ruby/3.2.0/lib/ruby/gems/3.2.0/gems/railties-7.0.4.3/lib/rails/commands/server/server_command.rb:5:in `<main>'
08:15:55 web.1  | /home/vscode/.asdf/installs/ruby/3.2.0/lib/ruby/gems/3.2.0/gems/bootsnap-1.16.0/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:32:in `require'
08:15:55 web.1  | /home/vscode/.asdf/installs/ruby/3.2.0/lib/ruby/gems/3.2.0/gems/bootsnap-1.16.0/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:32:in `require'
08:15:55 web.1  | /home/vscode/.asdf/installs/ruby/3.2.0/lib/ruby/gems/3.2.0/gems/railties-7.0.4.3/lib/rails/command/behavior.rb:44:in `block (2 levels) in lookup'
08:15:55 web.1  | /home/vscode/.asdf/installs/ruby/3.2.0/lib/ruby/gems/3.2.0/gems/railties-7.0.4.3/lib/rails/command/behavior.rb:40:in `each'
08:15:55 web.1  | /home/vscode/.asdf/installs/ruby/3.2.0/lib/ruby/gems/3.2.0/gems/railties-7.0.4.3/lib/rails/command/behavior.rb:40:in `block in lookup'
08:15:55 web.1  | /home/vscode/.asdf/installs/ruby/3.2.0/lib/ruby/gems/3.2.0/gems/railties-7.0.4.3/lib/rails/command/behavior.rb:39:in `each'
08:15:55 web.1  | /home/vscode/.asdf/installs/ruby/3.2.0/lib/ruby/gems/3.2.0/gems/railties-7.0.4.3/lib/rails/command/behavior.rb:39:in `lookup'
08:15:55 web.1  | /home/vscode/.asdf/installs/ruby/3.2.0/lib/ruby/gems/3.2.0/gems/railties-7.0.4.3/lib/rails/command.rb:73:in `find_by_namespace'
08:15:55 web.1  | /home/vscode/.asdf/installs/ruby/3.2.0/lib/ruby/gems/3.2.0/gems/railties-7.0.4.3/lib/rails/command.rb:46:in `invoke'
08:15:55 web.1  | /home/vscode/.asdf/installs/ruby/3.2.0/lib/ruby/gems/3.2.0/gems/railties-7.0.4.3/lib/rails/commands.rb:18:in `<main>'
08:15:55 web.1  | /home/vscode/.asdf/installs/ruby/3.2.0/lib/ruby/gems/3.2.0/gems/bootsnap-1.16.0/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:32:in `require'
08:15:55 web.1  | /home/vscode/.asdf/installs/ruby/3.2.0/lib/ruby/gems/3.2.0/gems/bootsnap-1.16.0/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:32:in `require'
08:15:55 web.1  | bin/rails:4:in `<main>'
08:15:55 web.1  | 3000
08:15:55 web.1  | exited with code 0
08:15:55 job.1  | terminated by SIGTERM
```
</p>
</details></div>

**Comment by icyleaf** [comment]
Fixed #1184

**Comment by InjoyDeng** [comment]
Add `RUN npm install -g sass esbuild` to the `.devcontainer/Dockerfile`.
